### PR TITLE
Added tests for generating sphinx documentation

### PR DESCRIPTION
**Pull request by BvB93**
## New
  * Added tests for generating the Sphinx documentation.

## Changed
  * Replaced ``requirements.txt`` with ``.readthedocs.yml``.
  * Fixed the jupyter notebook in the documentation.

### CODE REVIEW

File: .readthedocs.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# .readthedocs.yml
```
Comment (BvB93) [inline .readthedocs.yml:1]: In my experience a `.readthedocs.yml` file is generally less error-prone than `requirements.txt`.

**Comment by BvB93** [comment]
Wow, `nbsphinx` is one hell of a mess.

**Comment by BvB93** [comment]
Ok, it's finally all working.

File: .github/workflows/pythonapp.yml [hunk 2]
```diff
@@ -24,8 +24,8 @@ jobs:
     - run: conda --version
     - run: python --version
     - run: |
-        conda install -c conda-forge h5py rdkit
- 
+        conda install -c conda-forge h5py rdkit nbsphinx pandoc ipython
```
Comment (BvB93) [inline .github/workflows/pythonapp.yml:27]: Installing `nbsphinx` was suprinsgly tricky.
It seems that if you're using conda then you **will** have to install `nbsphinx` and the likes via conda as well.

Comment (felipeZ) [reply .github/workflows/pythonapp.yml:27]: Wouldn't be better to directly install `jupyter` install of ipython?

File: .readthedocs.yml (hunk 1 above)
Comment (felipeZ) [reply .readthedocs.yml:1]: I agree

File: setup.py [hunk 3]
```diff
@@ -13,6 +13,26 @@ def readme():
     with open('README.rst') as f:
         return f.read()
 
+docs_require = [
+    'sphinx>=2.1',
+    'sphinx-autodoc-typehints',
+    'sphinx_rtd_theme',
+    'nbsphinx',
+    'pandoc',
+    'ipython'
```
Comment (felipeZ) [inline setup.py:22]: I think that Jupyter will install ipython

File: test/test_sphinx.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
+"""Test the :mod:`sphinx` documentation generation."""
```
Comment (felipeZ) [inline test/test_sphinx.py:1]: Thanks for adding this test!

File: setup.py (hunk 3 above)
Comment (BvB93) [reply setup.py:22]: Huh, In fact both `pandoc` and `ipython` seem to be (indirect) `Jupyter` dependencies. 
Let's give it a try!

**Pull request merged**